plugins/statistics.koplugin/main.lua

```diff
@@ -175,14 +180,65 @@ function ReaderStatistics:initData()
     -- Update these numbers to what's actually stored in the settings
     -- (not that "notes" is invalid and does not represent edited highlights)
     self.data.highlights, self.data.notes = self.ui.bookmark:getNumberOfHighlightsAndNotes()
-    self.curr_total_time = 0
-    self.curr_total_pages = 0
     self.id_curr_book = self:getIdBookDB()
     self.total_read_pages, self.total_read_time = self:getPageTimeTotalStats(self.id_curr_book)
     if self.total_read_pages > 0 then
         self.avg_time = self.total_read_time / self.total_read_pages
     else
-        self.avg_time = 0
+        -- NOTE: Possibly less weird-looking than initializing this to 0?
+        self.avg_time = math.floor(0.50 * self.page_max_read_sec)
+        logger.info("ReaderStatistics: Initializing average time per page at 50% of the max value, i.e.,", self.avg_time)
+    end
+end
+
+-- Reset the (volatile) stats on page count changes (e.g., after a font size update)
+function ReaderStatistics:onUpdateToc()
+    local new_pagecount = self.view.document:getPageCount()
+
+    if new_pagecount ~= self.data.pages then
+        -- NOTE: insertDB does the exact same check
+        if not self.id_curr_book or self.mem_read_pages < MIN_PAGES_READ_TO_FLUSH then
+            logger.info("ReaderStatistics: Pagecount change, clearing volatile book statistics")
+            -- Clear volatile stats
+            self:resetVolatileStats(TimeVal:now().sec)
```

Comment:
Nope, it's the exact same check as insertDB, it aborts early below MIN_PAGES_READ_TO_FLUSH (i.e., 2).

This just ensures the collected data is dropped, too, because it'll have a bogus pagecount.

Reply:
Oh, ok, got confused, thought it was the other variable with 50.